rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/compile_time_interpreter.kt

```diff
@@ -139,36 +139,64 @@ open class BaseCompileTimeInterpreter(
             val field = it.fields.find { it.name.equals(declName, ignoreCase = true) }
             if (field != null) return (field.elementSize() /*/ field.declaredArrayInLine!!*/)
         }
-        rContext.statement()
```

Comment:
Ignore my comment, sorrry...

Reply:
I unresolve this conversation, because in a single commit you refactor and change, my suggestions are still valid.

Reply:
I am not sure if I understood your doubts about these changes but I'll try to explain them as much as possible. I going to describe each commit below. 

- [add T02_A50_P07 and T02_A50_P08 user cases] (https://github.com/smeup/jariko/pull/422/commits/dca03819fd4ea9fa9b588a66345ef7f1023f3c50):

As the comment says, I just added the user case test

- [add recursive search of variable size and variable type in case of LIKE expression](https://github.com/smeup/jariko/pull/422/commits/bd9b0f9e689d9ed24ec454537d4025b8ada94bff)

As the comment says, I moved the entire business logic of the case inside a new function which, can be invoked recursively. This function gets called when jariko has to loop through the statements to find where a specific variable has been defined.

Before the changes the structure was following:

```
open fun evaluateElementSizeOf(rContext: RpgParser.RContext, declName: String, conf: ToAstConfiguration): Int {

// look for the declaration of a variable in existing data definitions
knownDataDefinitions.forEach {
          ... 
        }

// if jariko does not return a value above, look for the the declaration in all statements (meaning all the lines of the RPGLE source file)

rContext.statement()
                .forEach {
                    when {
                        // read D spec
                        it.dspec() != null -> {
                              ..... 
                              return ...
                        }
                        // read C spec
                        it.cspec_fixed() != null -> {
                             .....
                             return ...
                        }
                       // read DS spec
                        it.dcl_ds() != null -> {
                            .....
                            return ...
``` 
The code above, did not consider the case of "block statements" like: IF, DOW, FOR, SELECT. These kind of statement, are composite statements, which means they include: C spec. Therefore, the solution was to create a recursive function named "findType". I moved the entire business logic inside this function. In case of "block statements", I just make the program invoke "findType" recursively so that it can loop through the C statements inside: IF, DOW, FOR, SELECT. You can see the new structure below:

// look for the declaration of a variable in existing data definitions
knownDataDefinitions.forEach {
          ... 
        }

// if jariko does not return a value above, look for the the declaration in all statements (meaning all the lines of the RPGLE source file)

rContext.statement()
                .forEach {
                    when {
                        // read D spec
                        it.dspec() != null -> {
                              ..... 
                              return ...
                        }
                        // read C spec
                        it.cspec_fixed() != null -> {
                             .....
                             return ...
                        }
                       // read DS spec
                        it.dcl_ds() != null -> {
                            .....
                            return ...
                     // read "block statements"
                      it.block() != null -> {
                           val size = findSize(...)
                            if (size != null) return size
                      }
``` 

As you can see, I just added the following "when" case: 
 
```
                     // read "block statements"
                      it.block() != null -> {
                           val size = findSize(...)
                            if (size != null) return size
                      }

```
In this case, we invoke the function "findSize" in order to loop inside the block statements. Since it's not sure we are going to find the declaration of the variable inside the block statement, before to return the value, we check whether the size is null. If null, we continue to loop through the other statements, otherwise we'll return the declaration. 

- [add recursive search of variable size and variable type in case of LIKE](https://github.com/smeup/jariko/pull/422/commits/baac155fe1fa146075e8a196e48bdd599daf16c7)

In this commits, as it says the description, I did the same thing described in the previous commit but with the function "findType". In fact, jariko invoke two different function to find size and type of the variable. So I used the same recursive approach. 

I hope this clear up the goal of the commits.